### PR TITLE
LLVM and SPIRV-LLVM-Translator pulldown (WW25)

### PR DESCRIPTION
**Pull request by sys-ce-bb**
LLVM: llvm/llvm-project@af35be5
SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@9823690

Community has started to refactor some code assuming opaque pointer only,
however the SPIRV component used in sycl is not ready to enable the opaque pointer by default,
so we have to revert some of the commit by guarding them within macro INTEL_SYCL_OPAQUEPOINTER_READY,
and keep both version of the code. 
The old code should be removed once we enable opaque pointer by default.

### CODE REVIEW

**Review by bader — DISMISSED** [review]
> LLVM: [llvm/llvm-project@af35be5](https://github.com/llvm/llvm-project/commit/af35be5) SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@

@sys-ce-bb, the translator commit is missing in the description. 
Please, resolve merge conflicts.

Did community change 1.5 M lines of code in two weeks? This seems too much compared to previous pulldowns.

**Comment by jsji** [comment]
> > LLVM: [llvm/llvm-project@af35be5](https://github.com/llvm/llvm-project/commit/af35be5) SPIRV-LLVM-Translator: KhronosGroup/SPIRV-LLVM-Translator@
> 
> @sys-ce-bb, the translator commit is missing in the description. Please, resolve merge conflicts.

Fixed.

> 
> Did community change 1.5 M lines of code in two weeks? This seems too much compared to previous pulldowns.

Yes. We have quite a few big changes in last two weeks, namely: refactoring of clang CodeGen to remove opaque pointer related code, removal of legacy passes.

**Comment by bader** [comment]
@jsji, thanks! Do you know who can help with resolving merge conflicts and fixing CI checks?

**Comment by bader** [comment]
@Chenyang-L, the project doesn't compile. Could you make sure that build and tests are passing, please?

**Comment by aelovikov-intel** [comment]
Skipped Windows testing is being addressed at https://github.com/intel/llvm/pull/10079.

File: clang/lib/CodeGen/CodeGenTypes.cpp
Comment (bader) [inline clang/lib/CodeGen/CodeGenTypes.cpp:128]: @jsji, it would be great to clarify the commit message the reason to revert the community patch. It's not obvious why do we need to keep these changes in our branch.

Comment (jsji) [reply clang/lib/CodeGen/CodeGenTypes.cpp:128]: It is not the first time we guard the revert , but sure, added in commit message.

**Comment by jsji** [comment]
@Chenyang-L  What is holding this pulldown?

**Comment by bader** [comment]
@Chenyang-L, please, fix CI checks.

**Comment by Chenyang-L** [comment]
> @Chenyang-L, please, fix CI checks.

@bader I don't think there are any outstanding fails from last week's checks that should be blocking. Latest run fails for unrelated reasons.

**Comment by bader** [comment]
> Latest run fails for unrelated reasons. What specifically do you see is blocking?

Could you clarify what are the reasons? 
I expect only two checks to fail: 
- SYCL Nightly / ubuntu2204_opaque_pointers_build_test / Build + LIT (pull_request)
- SYCL Pre Commit / lint (pull_request_target)

All other check pass in current CI, so this PR seems to introduce regressions. They must be addressed before PR can be merged.

**Pull request closed without merging**
